### DISCUSSION TITLE
Java  udf  performance  too  slow

### DISCUSSION THREAD

**Discussion by CoderXionghs** [apache/doris#23643] [A - General / Q&A]
I  wrote  a  Java  udf  to do  dic  trans，and  the  dic info  stores in  a sql database，so  in  my  Java  udf  code，i  do  the  initialize in  the  static code 。for  example

public  class  UdfClass {
      private  static  Map<string,string＞ dicMap;
      static{
           //initialize  dicMap  from  sql  database
      }

      public  string  evaluate (String  code){ 
            return  dicMap.get(code);
     }
}

i  test  the  performance，it's  too  slow。because  every  sql  function  call，this  udf  class  will  load once，and  reintialize。
    I wonder，why  not  this  udf  class should  always  stay  in  jvm  。

**Comment by LemonLiTree** [comment]
which doris version？

**Reply by CoderXionghs** [reply]
doris  2.0.0-rc04

**Reply by lsy3993** [reply]
Maybe we can get in touch，detail in wechat : lsy3993
